src/modules/components/proposal/proposalActions/proposalActions/proposalActions.tsx

```diff
@@ -54,30 +58,46 @@ export const ProposalActions: React.FC<IProposalActionsProps> = (props) => {
 
     return (
         <Card className={classNames('w-full overflow-hidden', className)}>
-            <Heading size="h2" className="p-4 md:p-6" ref={actionsContainerRef}>
-                {copy.proposalActionsContainer.containerName}
-            </Heading>
-            <Accordion.Container isMulti={true} value={expandedItems} onValueChange={handleAccordionValueChange}>
-                {actions.map((action, index) => (
-                    <ProposalActionsAction
-                        key={`action-${index}`}
-                        action={action}
-                        index={index}
-                        name={actionNames?.[action.type]}
-                        CustomComponent={customActionComponents?.[action.type]}
-                        {...web3Props}
+            {heading && (
+                <Heading size="h2" className="border-b border-neutral-100 p-4 md:p-6">
+                    {heading}
+                </Heading>
+            )}
+            <Accordion.Container
+                ref={actionsContainerRef}
+                isMulti={true}
+                value={expandedItems}
+                onValueChange={handleAccordionValueChange}
+            >
+                {actions.length > 0 ? (
+                    actions.map((action, index) => (
+                        <ProposalActionsAction
+                            key={`action-${index}`}
+                            action={action}
+                            index={index}
+                            name={actionNames?.[action.type]}
+                            CustomComponent={customActionComponents?.[action.type]}
+                            {...web3Props}
+                        />
+                    ))
+                ) : (
+                    <EmptyState
+                        heading={copy.proposalActionsContainer.empty.heading}
+                        description={copy.proposalActionsContainer.empty.description}
+                        isStacked={false}
+                        objectIllustration={{ object: 'SMART_CONTRACT' }}
                     />
-                ))}
-                <div className="mt-1 flex w-full flex-col gap-y-3 px-4 pb-4 md:flex-row-reverse md:px-6 md:pb-6">
-                    {actions.length > 1 && (
+                )}
+                {actions.length > 1 && (
+                    <div className="mt-1 flex w-full flex-col gap-y-3 px-4 pb-4 md:flex-row-reverse md:px-6 md:pb-6">
                         <Button onClick={handleToggleAll} variant="tertiary" size="md" className="shrink-0 md:w-fit">
                             {expandedItems.length === actions.length
                                 ? copy.proposalActionsContainer.collapse
                                 : copy.proposalActionsContainer.expand}
                         </Button>
-                    )}
-                    {children}
-                </div>
+                    </div>
+                )}
+                {children}
```

Comment:
We'll probably need to keep the `children` inside the wrapper div to render the `execute actions` button inline with the toggle-all button, right?